src/librustc/hir/map/mod.rs

```diff
@@ -608,6 +609,80 @@ impl<'hir> Map<'hir> {
         &self.forest.krate.attrs
     }
 
+    pub fn visit_module_item_likes<V>(&self, module: DefId, visitor: V)
+        where V: itemlikevisit::ItemLikeVisitor<'hir>
+    {
+        let node_id = self.as_local_node_id(module).unwrap();
+
+        // Read the module so we'll be re-executed if new items appear in the module
+        self.read(node_id);
```

Comment:
This doesn't actually work as intended yet (see https://github.com/rust-lang/rust/issues/40876). We should try and see if we can make this work properly.

Reply:
I was thinking of adding a `HIRModule` dep-node. That would reduce the dep nodes used by the passes too.

Reply:
> I was thinking of adding a HIRModule dep-node.

So that the hash of the `HIRModule` dep-node would be a combination of the hashes of the module's items?

Reply:
Yeah.